platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -110,9 +151,84 @@ def get_firmware_version(self):
         return result
 
 
+    def _check_onie_version(self):
+        # check ONIE version. To update ONIE requires version 5.2.0016 or later.
+        machine_conf = self._read_generic_file(MACHINE_CONF_FILE, MACHINE_CONF_MAX_SIZE)
+
+        try:
+            machine_info = get_machine_info()
+            onie_version_string = machine_info['onie_version']
```

Comment:
@stephenxs please use a named definition for 'onie_version'

Reply:
No, I don't think we need that. If some literal constant is referenced frequently we need to define constant for it. But if it's common and referenced only once we don't need to.

Reply:
@stephenxs ok.